### PR TITLE
[addons][platform] cleanup API, improve "C" interface, documentation rework for Doxygen

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description

Make this addon interface like the others usable within "C" and make his docs usable again on addon doxygen view.

Change here only relates to inputstream.adaptive but not need a version increase on addon.

## User related:

Not user relevant

Only for all binary addon system requests may be for the user:
- Improvement: Binary addon system revised to ensure higher security in its data exchange between addon and Kodi
- Improvement: Revised binary addon system documentation for outside developers

Commit 1:
--------------------------
**[addons][platform][android] spearate "C" parts to own file**

This place the "C" parts about game to a new file on "kodi/c-api/platform/android/system.h".
This done to allow also "C" language only (as base for other languages) and to confirm safe "C" ABI
between addon and Kodi.

Commit 2:
--------------------------
**[addons][platform] change documentation to have more usable**

Before was it not good viewed inside doxygen generated HTML as the
functions was viewed outside of class name.

This change to define class name and increase a bit platform specific group descriptions.

Also was wrongly before a empty `@param[in]` used where brings conflict to coming `@return`

On the API itself is by this and commit before nothing in code work changed and not needs new versions.

Commit 3:
--------------------------
**[addons][platform][android] increase min API to 1.0.2**

## Screenshots (if appropriate):
Pictures about his docs:
Bildschirmfoto vom 2020-09-12 11-26-41
Bildschirmfoto vom 2020-09-12 11-34-46

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
I bring it in now, as it is only a minor type of changes.

File: xbmc/addons/kodi-dev-kit/include/kodi/c-api/platform/android/system.h [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 #define INTERFACE_ANDROID_SYSTEM_NAME "ANDROID_SYSTEM"
 #define INTERFACE_ANDROID_SYSTEM_VERSION "1.0.1"
-#define INTERFACE_ANDROID_SYSTEM_VERSION_MIN "1.0.1"
+#define INTERFACE_ANDROID_SYSTEM_VERSION_MIN "1.0.2"
```
Comment (ksooo) [inline xbmc/addons/kodi-dev-kit/include/kodi/c-api/platform/android/system.h:16]: @AlwinEsch  How could INTERFACE_ANDROID_SYSTEM_VERSION_MIN (1.0.2) be greater than INTERFACE_ANDROID_SYSTEM_VERSION (1.0.1) ? I'm pretty sure this leads to crashes...

**Review by peak3d — COMMENTED** [review]
lgtm

**Pull request merged**